src/par/FixColumnPartitions.c

```diff
@@ -19,134 +19,166 @@
 extern "C" {
 #endif
 
-int FixColumnPartitions(Mesh_ptr mesh, int *part, MSTK_Comm comm) { 
-  int i, j, k, idx, nrf, nfr, nfv, nrv, curid, nxtid, homepid, nxtpid;
-  int interior, done, nmove=0, found;
-  MRegion_ptr mr, curreg, nxtreg;
-  MFace_ptr rf, topf, botf;
-  MVertex_ptr fv;
-  List_ptr rfaces, fregs, fverts;
-  double fxyz[MAXPV2][3], rxyz[MAXPV3][3], rfvec[3], rcen[3], fcen[3], maxdp, norm;
+int FixColumnPartitions_IsSide(Mesh_ptr mesh, MRegion_ptr mr, MFace_ptr rf) {
+  double fxyz[MAXPV2][3];
+  double znorm;
+  int j, jm, jp, nfv;
   
-  if (MESH_Num_Regions(mesh) == 0) return 1; /* Not implemented for 2D (perhaps not needed either) */
+  MF_Coords(rf,&nfv,fxyz);
 
-  idx = 0;
-  while ((mr = MESH_Next_Region(mesh,&idx))) {
-    rfaces = MR_Faces(mr);
-    nrf = List_Num_Entries(rfaces);
-    
-    interior = 1;
-    for (i = 0; i < nrf; i++) {
-      rf = List_Entry(rfaces,i);      
-      fregs = MF_Regions(rf);
-      if (List_Num_Entries(fregs) == 1) 
-        interior = 0;
-      List_Delete(fregs);
-      if (!interior) break;
-    }
+  znorm = 0.;
+  for (j = 0; j < nfv; j++) {
+    /* determine the average z-normal of the face */
+    if (j == 0) jm = nfv-1;
+    else jm = j-1;
 
-    if (interior) {
-      List_Delete(rfaces);
-      continue;
-    }
+    if (j == nfv-1) jp = 0;
+    else jp = j+1;
+
+    znorm += (fxyz[j][1]-fxyz[jm][1]) * (fxyz[jp][0] - fxyz[j][0])
+        - (fxyz[jp][1]-fxyz[j][1]) * (fxyz[j][0] - fxyz[jm][0]);
+  }
+  znorm /= nfv;
+  if (fabs(znorm) < 1.e-6) 
+    /* z-normal is zero, face is a column side by definition */
+    return 1;
```

Comment:
Are you absolutely sure you will never need to use a mesh with wavy columns where the sides are not vertical?